### PR TITLE
Move passes implementation to GlobalOptimization.

### PR DESCRIPTION
**Pull request by hanhanW**
These passes are not related to any flow specific logics, and they are only used by GlobalOptimization. So we move them to GlobalOptimization/. The passes are:

- Convert1X1FilterConv2DToMatmul
- DetachElementwiseFromNamedOps
- EraseUnusedLinalgOperands
- RemoveZeroExtentTensors
- SetEncoding

### CODE REVIEW

**Comment by hanhanW** [comment]
cc @dcaballe @pzread @NatashaKnk who might be working on data-tiling for matvec, etc (https://github.com/openxla/iree/issues/15053)

**Comment by NatashaKnk** [comment]
Thanks for the heads up!

File: compiler/src/iree/compiler/GlobalOptimization/Passes.td [hunk 1]
```diff
@@ -9,6 +9,35 @@
 
 include "mlir/Pass/PassBase.td"
 
+def Convert1X1FilterConv2DToMatmul:
```
Comment (MaheshRavishankar) [inline compiler/src/iree/compiler/GlobalOptimization/Passes.td:12]: Nit: could we make the passes alphabetically ordered here.

**Review by MaheshRavishankar — APPROVED** [review]
Thanks!

**Pull request merged**